### ISSUE TITLE
Error generating Ku from authentication pass phrase.

### ISSUE THREAD

**Issue by APT256** [kko/synology-diskstation-zabbix-snmpv3#4]
I just installed your template into Zabbix 4.4. I have a Synology DS416 NAS that has SNMP v3 configured and enabled on it. I imported your template and then created a new host for the Synology NAS using the template. I then created these 3 macros:
{$SNMP_USERNAME}
{$SNMP_AUTHPASS}
{$SNMP_PRIVPASS}

I then entered the values for the macros by cut/paste (so there would be no typos). After about 30 minutes, the SNMP availability indicator for the Synology NAS is red and if I click on it, it gives this error: "Error generating Ku from authentication pass phrase."

The only thing I can think of that might be causing a problem are the pass erases. I am using long (20 character), complex (upper, lower, number, symbol) as well. Do think that this could be the cause of my problem?

Thanks!
Jon

**Comment by kko** [comment]
Did you select AES for privacy protocol and SHA for username/pw protocol?
You may also double check if you did not mix up the two passwords

**Comment by APT256** [comment]
I went back and double-checked everything, and I'm pretty sure that I have it set correctly. I even to the cli and did a snmpwalk with the v3 creds and I received a lot of data back.

**Comment by APT256** [comment]
So this is odd... I had to rebuild my Zabbix server as the old hard drive was too slow and generating a lot of errors. Anyway, I rebuilt everything from scratch, and now the template appears to be working fine. The only thing I did differently this time was to run and snap v3 snmpwalk on the NAS before creating the client. I know it shouldn't matter, and maybe it doesn't, but it is working now while it didn't work yesterday.
